### PR TITLE
Clean up VS experience for s.p.corelib.csproj

### PR DESCRIPTION
**Pull request by danmosemsft**
VS .shproj shared files project does not work correctly when the files shared are below the folder of the .csproj. Since we aren't loading the other corelib csproj into the solution, it's really easier to just get rid of the .shproj. As it's an import, mark the files visible in the solution explorer.

Move AssemblyInfoFiles into the targets file, to stop VS trying to treat them as source files. Also remove the target in GenerateVersionInfo.targets which I saw was just overridden by the one in Tools\versioning.targets.

This cuts the warnings in VS down from "lots" to one -- a warning about BaseAddress being too large, which is a bug in VS. The base address is fine, and is consumed in the build correctly. I opened a bug against VS.

### CODE REVIEW

**Comment by danmosemsft** [comment]
I diffed the dll before and after to check the assembly attributes are the same

File: src/mscorlib/System.Private.CoreLib.sln [hunk 1]
```diff
@@ -1,17 +1,11 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
-VisualStudioVersion = 15.0.26208.0
+VisualStudioVersion = 15.0.26817.0
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "System.Private.CoreLib", "System.Private.CoreLib.csproj", "{3DA06C3A-2E7B-4CB7-80ED-9B12916013F9}"
 EndProject
-Project("{D954291E-2A0B-460D-934E-DC6B0785DB48}") = "System.Private.CoreLib.Shared", "shared\System.Private.CoreLib.Shared.shproj", "{C5ED3C1D-B572-46F1-8F96-522A85CE1179}"
```
Comment (jkotas) [inline src/mscorlib/System.Private.CoreLib.sln]: Delete `shared\System.Private.CoreLib.Shared.shproj` since it is not used anymore?

Comment (jkotas) [inline src/mscorlib/System.Private.CoreLib.sln]: (I assume there will be matching change in CoreRT to make it look the same there.)

File: src/mscorlib/shared/System.Private.CoreLib.Shared.projitems [hunk 2]
```diff
@@ -13,6 +13,11 @@
     <TargetsUnix Condition="'$(TargetsUnix)' != 'true'">false</TargetsUnix>
     <TargetsOSX Condition="'$(TargetsOSX)' != 'true'">false</TargetsOSX>
   </PropertyGroup>
+  <ItemDefinitionGroup>
+    <Compile>
+        <Visible>true</Visible>
```
Comment (jkotas) [inline src/mscorlib/shared/System.Private.CoreLib.Shared.projitems]: Nit: The rest of the file is on 2 space indentation plan.

File: src/mscorlib/System.Private.CoreLib.sln (hunk 1 above)
Comment (danmosemsft) [inline src/mscorlib/System.Private.CoreLib.sln]: sure I'll fix that also.

**Comment by danmosemsft** [comment]
@jkotas ok?

**Pull request merged**